esp-hal/src/uart.rs

```diff
@@ -642,25 +636,21 @@ where
     }
 
     /// Writes bytes
-    pub fn write_bytes(&mut self, data: &[u8]) -> Result<usize, Error> {
+    pub fn write_bytes(&mut self, data: &[u8]) -> usize {
```

Comment:
I'm a bit stumped on if this is correct or not. As far as I can tell, UART Tx seems to be infallible (in terms of detecting errors), so this change does make sense in a way. However, a user may pass in an empty buffer, which might be considered an invalid argument, but AFAIK, nothing bad would happen passing in an empty buffer and it won't violate any embedded-io restrictions: https://github.com/rust-embedded/embedded-hal/blob/9941a6468853c447c892ab64eb78109f0198910d/embedded-io/src/lib.rs#L394-L395

Reply:
I _think_ in some other PR we said a zero-length slice write shouldn't be an error (and we changed it there - can't remember where t.b.h.) - but at least we should be consistent with that

Reply:
Can we keep the errors here, and just make them empty for now? Technically, sending a byte can fail if the lines are broken, I'm just not sure if we can detect that, but it might be good to keep the option in the future.

Reply:
> I think in some other PR we said a zero-length slice write shouldn't be an error (and we changed it there - can't remember where t.b.h.) - but at least we should be consistent with that

I think that was I2C.

Reply:
Yeah my gut is saying we should return an error here, I think it's better to be safe just, even if we don't return any errors just yet.

Reply:
I just wanted to note that there's no need to follow e-io rules or semantics on the inherent APIs, especially since the crate hasn't stabilised, and may change after esp-hal is stabilised.